packages/@storefront/core/src/core/url-beautifier/details/parser.ts

```diff
@@ -18,6 +18,13 @@ export class DetailsUrlParser extends UrlParser<UrlBeautifier.DetailsUrlState> {
     const title = path.shift();
     const id = path.pop();
     let variants = [];
+    const queryParams: {[key: string]: string} = {};
+
+    if (Object.keys(uri.query).length) {
+      const filteredQueryParams = DetailsUrlParser.extractFilteredQueryParams(uri.query);
+      queryParams.area = filteredQueryParams.area;
+      queryParams.collection = filteredQueryParams.collection;
```

Comment:
These should be guarded to check for `undefined`. You don't want to add it if it doesn't exist.

Reply:
+1 for this.

Reply:
Agreed, will changed.